apps/gitness/src/pages-v2/pull-request/__tests__/pull-request-utils.test.ts

```diff
@@ -0,0 +1,253 @@
+    ]
+  }
+}
+
+const mockReviewDecision = 'approved' as EnumPullReqReviewDecision
+const mockReviewedSHA = 'sha1'
+const mockSourceSHA = 'sha2'
+
+// Mock data for findWaitingDecisions
+const mockEntriesForWaitingDecisions: TypesCodeOwnerEvaluationEntry[] = [
+  {
+    owner_evaluations: [
+      { review_decision: undefined }, // Empty decision
+      { review_decision: 'request_changes' as EnumPullReqReviewDecision }
+    ]
+  },
+  {
+    owner_evaluations: [{ review_decision: 'approved' }]
+  },
+  {
+    owner_evaluations: [
+      { review_decision: undefined }, // Empty decision
+      { review_decision: 'reviewed' as EnumPullReqReviewDecision }
+    ]
+  }
+]
+
+// Mock data for findChangeReqDecisions
+const mockEntriesForChangeReq: TypesCodeOwnerEvaluationEntry[] = [
+  {
+    owner_evaluations: [
+      { review_decision: 'approve' as EnumPullReqReviewDecision },
+      { review_decision: 'request_changes' as EnumPullReqReviewDecision }
+    ]
+  },
+  {
+    owner_evaluations: [{ review_decision: 'approve' as EnumPullReqReviewDecision }]
+  },
+  {
+    owner_evaluations: [{ review_decision: 'reviewed' as EnumPullReqReviewDecision }]
+  }
+]
+// Mock data for extractInfoFromRuleViolationArr
+const mockRuleViolations = [
+  {
+    violations: [{ message: 'Violation 1' }, { message: 'Violation 2' }]
+  },
+  {
+    violations: [{ message: 'Violation 3' }]
+  }
+]
+
+// Tests
+describe('processReviewDecision', () => {
+  it('should mark review as outdated if SHAs differ', () => {
+    const result = processReviewDecision(mockReviewDecision, mockReviewedSHA, mockSourceSHA)
+    expect(result).toBe(PullReqReviewDecision.outdated)
+  })
+})
+
+describe('generateStatusSummary', () => {
+  it('should generate a summary of status checks', () => {
+    const result = generateStatusSummary(mockChecks)
+    expect(result.summary.total).toBe(2)
+    expect(result.summary.successReq).toBe(1)
+    expect(result.summary.failed).toBe(1)
+  })
+})
+
+// Mock data for determineStatusMessage
+const mockChecksForStatusMessage = [
+  { check: { status: 'success' as EnumCheckStatus }, required: true },
+  { check: { status: 'failure' as EnumCheckStatus }, required: false }
+]
+
+describe('determineStatusMessage', () => {
+  it('should determine the status message based on the status summary', () => {
+    const result = determineStatusMessage(mockChecksForStatusMessage as TypeCheckData[])
+    expect(result).toEqual({
+      color: 'text-success',
+      content: '1 failed, 1 succeeded',
+      status: 'success',
+      title: 'All required checks passed'
+    })
+  })
+})
+
+describe('extractSpecificViolations', () => {
+  it('should extract specific violations based on rule', () => {
+    const result = extractSpecificViolations(mockViolationsData, 'rule1')
+    expect(result).toEqual([{ code: 'rule1', description: 'Violation 1' }])
+  })
+})
+
+describe('checkIfOutdatedSha', () => {
+  it('should return true if SHAs differ', () => {
```

Comment:
Would be good to prove when it would return `false` as well.

Reply:
cant actually return false, only outdated or the original review decision that it was passed to